docker/run_on_docker.sh

```diff
@@ -85,23 +85,34 @@ bootstrap_image() {
   rm -fR $build_workdir
 }
 
+push_image() {
+    local image_name="$1"
+    docker push $image_name || die "Failed to push docker image $image_name"
```

Comment:
That would be perfect.  Failure as a default is correct, I think, but we might want to overwrite it with an environment variable.